decentraland/redux/middlewares.js

```diff
@@ -0,0 +1,60 @@
+import { applyMiddleware } from "redux";
+import {
+  SET_ACCOUNT,
+  SET_ACCOUNT_CREATION_STATUS,
+  UPDATE_ACTION_FOR_ACCOUNT_CREATION_STATUS,
+  SET_ACCOUNT_CREATION_ACTIONS,
+  ADD_TO_MY_ASSETS_LIST,
+  REMOVE_MY_ASSET_FROM_LIST,
+  SET_MY_ASSETS_LIST,
+} from "./actions";
+import {
+  storeAccount,
+  storeAccountCreationStatus,
+  storeAccountCreationActions,
+  storeMyAssets,
+} from "@helpers/storage";
+
+const storer = store => next => async action => {
+  const { type } = action;
+  const result = next(action);
+  const nextState = store.getState();
+  const { accountInfo, myAssets } = nextState;
+  const { account, creationStatus, creationActions } = accountInfo;
+  const { list: myAssetsList } = myAssets;
+
+  switch (type) {
+    case SET_ACCOUNT: {
+      await storeAccount(account);
+      break;
+    }
+    case SET_ACCOUNT_CREATION_STATUS: {
+      await storeAccountCreationStatus(creationStatus);
+      break;
+    }
+    case UPDATE_ACTION_FOR_ACCOUNT_CREATION_STATUS: {
+      await storeAccountCreationActions(creationActions);
+      break;
+    }
+    case SET_ACCOUNT_CREATION_ACTIONS: {
+      await storeAccountCreationActions(creationActions);
+      break;
+    }
+    case ADD_TO_MY_ASSETS_LIST: {
+      await storeMyAssets(myAssetsList);
+      break;
+    }
+    case REMOVE_MY_ASSET_FROM_LIST: {
+      await storeMyAssets(myAssetsList);
+      break;
+    }
+    case SET_MY_ASSETS_LIST: {
+      await storeMyAssets(myAssetsList);
+      break;
+    }
```

Comment:
(Comment on somewhat random line) Do we need any error handling code in this middleware?

Reply:
- For action that was dispatched manually (those from `App` component), we can catch it and show an error on UI.
- Not sure if it will be possible to do that (show on UI) from actions dispatched by redux.

Reply:
`Not sure if it will be possible to do that (show on UI) from actions dispatched by redux.`

What's an example of an action dispatched by redux that "we" don't trigger?

Reply:
Maybe I'm wrong, but I've undestood from the docs that 

`store.dispatch(setAccountCreationStatus(creationStatus));` (App.js) will throw error from middleware but `setAccountCreationStatus(creationStatus)` (EthereumSignUpScreen.js) won't.

Reply:
> Maybe I'm wrong, but I've undestood from the docs that
> 
> `store.dispatch(setAccountCreationStatus(creationStatus));` (App.js) will throw error from middleware but `setAccountCreationStatus(creationStatus)` (EthereumSignUpScreen.js) won't.

I understand now. That makes sense to me. I think you're right about this. So for cases where we can't throw, we still might want to log the error from within the middleware.

Reply:
OK, should I add some `console.log` commands thru the middleware?

Reply:
I think so, if that works for you. I forget, do we have tooling in place to strip console.log for production builds yet?

Reply:
We have it on babel config but have to be done manually.

Reply:
What has to be done manually?

Reply:
Uncomment `plugins: ["transform-remove-console"],` from `babel.config.js`.

Reply:
Ah ok. Any reason not to have it always enabled?

Reply:
If we want to have the app ready for a develop env, make sense make that way otherwise no debug message/error will be showed on the console.

Reply:
Oh right. So what we really want to do is write a little bit of logic that enables or disables this based on env.

Reply:
Or more likely write our own tiny logger helper that does that same thing without using the babel feature

Reply:
https://github.com/tasitlabs/tasit/pull/323

Reply:
Awesome, thanks! So if we want to go back and add a console.log here in the middleware, we can now